match_minima.py

```diff
@@ -0,0 +1,787 @@
+        else:
+            lowest_conf_indices.append(nan_cnt.index(min(nan_cnt)))
+
+    # after going thru all mols, calc relative energies by subtracting lowest
+    rel_energies = []
+    for z, molE in zip(lowest_conf_indices, matched_enes):
+
+        # temp list for this mol's relative energies
+        temp = []
+
+        # subtract energy of lowest_conf_index; store in list
+        for fileE in molE:
+            temp.append(
+                [(fileE[i] - fileE[z]) for i in range(len(fileE))])
+        rel_energies.append(temp)
+
+    return rel_energies, lowest_conf_indices
+
+
+def write_rel_ene(mol_name, rmse, relEnes, low_ind, ff_list, prefix='relene'):
+    """
+    Write the relative energies and RMSEs in an output text file.
+
+    Parameters
+    ----------
+    mol_name : string
+        title of the mol being written out
+    rmse : list
+        1D list of RMSEs for all the compared methods against ref method
+    relEnes : 2D list
+        relEnes[i][j] represents energy of ith method and jth conformer
+    low_ind : int
+        integer of the index of the lowest energy conformer
+    ff_list : list
+        list of methods including reference as the first
+    prefix : string
+        prefix of output dat file
+
+    """
+
+    ofile = open(f"{prefix}_{mol_name}.dat", 'w')
+    ofile.write(f"# Molecule %s\n" % mol_name)
+    ofile.write("# Energies (kcal/mol) for conformers matched to first method.")
+    ofile.write(f"\n# Energies are relative to conformer {low_ind}.\n")
+    ofile.write("# Rows represent conformers; columns represent methods.\n")
+
+    # write methods, RMSEs, integer column header
+    rmsheader = "\n# "
+    colheader = "\n\n# "
+    for i, method in enumerate(ff_list):
+        ofile.write(f"\n# {i+1} {method}")
+        rmsheader += f"\t{rmse[i]:.4f}"
+        colheader += f"\t\t{i+1}"
+
+    ofile.write("\n\n# RMS errors by method, with respect to the " +
+                "first method listed:")
+    ofile.write(rmsheader)
+    ofile.write(colheader)
+
+    # write each ff's relative energies
+    for i in range(len(relEnes[0])):
+
+        # label conformer row
+        ofile.write(f"\n{i}\t")
+
+        # write energies for this conformer of all methods
+        thisline = [x[i] for x in relEnes]
+        thisline = ['%.4f' % elem for elem in thisline]
+        thisline = '\t'.join(map(str, thisline))
+        ofile.write(thisline)
+
+    ofile.close()
+
+
+def extract_matches(mol_dict):
+    """
+    This function checks if minima is matched, using indices lists inside dict.
+    If match is found, store the corresponding energy under a new key with
+    value of "energies_matched". If there is no match, the energy listed in the
+    dict is not used; rather, nan is added as placeholder.
+
+    Parameter
+    ---------
+    mol_dict : dict of dicts
+        mol_dict['mol_name']['energies']
+        = [[file1: conf1 conf2] [file2: conf1 conf2]]
```

Comment:
👍  Nice docstring for a complex structure.

Reply:
I don't quite understand the phrase "indices lists inside dict." though.